strings/test.cc

```diff
@@ -78,3 +89,123 @@ TEST(FixedSizeSerializer, ImplicitSyntax) {
     EXPECT_EQ("01000000000000000000", PackToString(x));
   }
 }
+
+static const char global_string[] = "magic";
+
+TEST(Util, CompileTimeStringLength) {
+  const char local_string[] = "foo";
+  static const char local_static_string[] = "blah";
+  EXPECT_EQ(3u, CompileTimeStringLength(local_string));
+  EXPECT_EQ(4u, CompileTimeStringLength(local_static_string));
+  EXPECT_EQ(5u, CompileTimeStringLength(global_string));
+}
+
+TEST(JoinAndSplit, Join) {
+  EXPECT_EQ("one,two,three", Join({"one", "two", "three"}, ','));
+  EXPECT_EQ("onetwothree", Join({"one", "two", "three"}, ""));
+  EXPECT_EQ("one, two, three", Join({"one", "two", "three"}, ", "));
+  EXPECT_EQ("one, two, three", Join({"one", "two", "three"}, std::string(", ")));
+  EXPECT_EQ("", Join({}, ' '));
+  EXPECT_EQ("", Join({}, " "));
+
+  EXPECT_EQ("a,b,c,b", Join(std::vector<std::string>({"a", "b", "c", "b"}), ','));
+  EXPECT_EQ("a,b,c", Join(std::set<std::string>({"a", "b", "c", "b"}), ','));
+  EXPECT_EQ("a,b,b,c", Join(std::multiset<std::string>({"a", "b", "c", "b"}), ','));
+}
+
+TEST(JoinAndSplit, Split) {
+  EXPECT_EQ("one two three", Join(Split("one,two,three", ','), ' '));
+  EXPECT_EQ("one two three four", Join(Split("one,two|three,four", ",|"), ' '));
+  EXPECT_EQ("one two three four", Join(Split("one,two|three,four", std::string(",|")), ' '));
+  EXPECT_EQ("one,two three,four", Join(Split("one,two|three,four", '|'), ' '));
+  EXPECT_EQ("one,two three,four", Join(Split("one,two|three,four", "|"), ' '));
+  EXPECT_EQ("one,two three,four", Join(Split("one,two|three,four", std::string("|")), ' '));
+
+  EXPECT_EQ("one two three", Join(Split(",,one,,,two,,,three,,", ','), ' '));
+  EXPECT_EQ("  one   two   three  ", Join(Split(",,one,,,two,,,three,,", ',', TrimMode::NoTrim), ' '));
+
+  EXPECT_EQ("one two three", Join(Split<ByWhitespace>("one two three"), ' '));
+  EXPECT_EQ("one two three", Join(Split<ByWhitespace>("\t \tone\t \ttwo\t \tthree\t \t"), ' '));
+}
+
+TEST(JoinAndSplit, FunctionalSplit) {
+  {
+    std::string result;
+    Split("one,two,three", ',', [&result](const std::string& s) { result += s + '\n'; });
+    EXPECT_EQ("one\ntwo\nthree\n", result);
+  }
+  {
+    std::string result;
+    Split("one,two,three", ',', [&result](std::string&& s) { result += s + '\n'; });
+    EXPECT_EQ("one\ntwo\nthree\n", result);
+  }
+  {
+    std::string result;
+    struct Helper {
```

Comment:
I don't see you use `helper` instead of a lambda. Either I don't see some C++ construct, or this is a copy-paste error. Same for the next test.